app/controllers/users/sessions_controller.rb

```diff
@@ -21,7 +21,7 @@ def new
       )
 
       @ial = sp_session ? sp_session_ial : 1
-      session[:ial2_request_with_no_sp] = true if sp_session.blank? && params[:ial] == '2'
+      session[:ial2_request_with_no_sp] = campaign if sp_session.blank? && params[:ial] == '2'
```

Comment:
Is there a reason we have to stuff the campaign in the session here? This really looks like a boolean. This may be easier to work with down the line if we did something like:

```ruby
if sp_session.blank? && params[:ial] == '2'
  session[:ial2_request_with_no_sp] = true
  session[:ial2_request_campaign] = params[:campaign]
end
```

Reply:
Ah.  Actually I wanted to change the name of ial2_request_with_no_sp to ial2_request_with_no_sp_campaign and the no campaign link will be some standard string like 'no campaign' which we still want to track